utils/docker/Dockerfile.maldet.centos.7

```diff
@@ -0,0 +1,101 @@
+                build_number="${branch#*:}";                              \
+                branch="${branch%:*}";                                    \
+            fi;                                                           \
+        fi;                                                               \
+        echo -e "[$repo:$branch:$build_number]\n\
+name=$repo:$branch:$build_number\n\
+baseurl=${JENKINS_URL}job/daos-stack/job/$repo/job/$branch/$build_number/artifact/artifacts/centos7/\n\
+enabled=1\n\
+gpgcheck=False\n" >> /etc/yum.repos.d/$repo:$branch:$build_number.repo;   \
+        cat /etc/yum.repos.d/$repo:$branch:$build_number.repo; \
+        dnf repolist; \
+        dnf --disablerepo=\* --enablerepo=$repo:$branch:$build_number makecache; \
+    done
+
+# Install OS updates and package.  Include basic tools and daos dependencies
+COPY ./utils/scripts/install-centos7.sh /tmp/install.sh
+RUN chmod +x /tmp/install.sh && dnf upgrade && /tmp/install.sh && dnf clean all && \
+    rm -f /tmp/install.sh
+
+ARG UID=1000
+
+# Add DAOS users
+RUN useradd --no-log-init --uid $UID --user-group --create-home --shell /bin/bash \
+            --home /home/daos daos_server
+RUN echo "daos_server:daos_server" | chpasswd
+RUN useradd --no-log-init --user-group --create-home --shell /bin/bash daos_agent
+RUN echo "daos_agent:daos_agent" | chpasswd
+RUN echo "daos_server ALL=(root) NOPASSWD: ALL" >> /etc/sudoers.d/daos_sudo_setup; \
+    chmod 0440 /etc/sudoers.d/daos_sudo_setup;                                     \
+    visudo -c;                                                                     \
+    sudo -l -U daos_server
+
+# Create directory for DAOS backend storage
+RUN mkdir -p /opt/daos /mnt/daos /var/run/daos_server /var/run/daos_agent /home/daos/pre /home/daos/daos &&   \
+    chown -R daos_server.daos_server /opt/daos /mnt/daos /var/run/daos_server /home/daos &&  \
+    chown daos_agent.daos_agent /var/run/daos_agent
+
+# Now do an update to ensure software is up to date for install.  If the
+# src hasn't changed then this won't do anything, but if it has then we want to
+# ensure that latest dependencies are used.
+RUN {                                                                                  \
+        dnf upgrade --exclude=spdk,spdk-devel,dpdk-devel,dpdk,mercury-devel,mercury && \
+        dnf clean all;                                                                 \
+    }
+
+# force an upgrade to get any newly built RPMs, but only if CACHEBUST is set.
+ARG CACHEBUST
+RUN [ -z "$CACHEBUST" ] || {                                                           \
+        dnf upgrade --exclude=spdk,spdk-devel,dpdk-devel,dpdk,mercury-devel,mercury && \
+        dnf clean all;                                                                 \
+    }
+
+ARG BUILD_URL
+COPY ci/rpm/install_maldet.sh /tmp/install_maldet.sh
+COPY ci/rpm/distro_info.sh /tmp/distro_info.sh
+RUN /tmp/install_maldet.sh
```

Comment:
To expand on my comments above I'd expect "/usr/local/sbin/maldet --update-sigs" and "freshclam" to be run as dedicated steps in the dockerfile, protected by appropriate CACHEBUST mechanism to ensure that updates are applied in-line with policy.

Reply:
The BUILD_URL for the job needs to be added to the repository of the dockerfile, so must be done after the CACHEBUST.

This also is the place to make a effort at updating the malware definitions.